scrapy/pipelines/images.py

```diff
@@ -115,7 +118,16 @@ def file_downloaded(self, response, request, info, *, item=None):
 
     def image_downloaded(self, response, request, info, *, item=None):
         checksum = None
-        for path, image, buf in self.get_images(response, request, info, item=item):
+        image_stream = self.get_images(response, request, info, item=item)
+        while True:
+            try:
+                path, image, buf = next(image_stream)
+            except OSError:
+                logger.exception('Could not process image')
+                continue
```

Comment:
This is implicit (unneeded), as the `except` will fall through.

Reply:
It would still execute the rest of the code within the loop, though, with references to the image data from the previous iteration. I think this is needed.

Reply:
Should I correct it or not?